web/src/style.css

```diff
@@ -1,3 +1,95 @@
+  --wp-control-ok-300: theme('colors.int-wp-control-ok.300');
+
+  --wp-control-error-100: theme('colors.int-wp-control-error.100');
+  --wp-control-error-200: theme('colors.int-wp-control-error.200');
+  --wp-control-error-300: theme('colors.int-wp-control-error.300');
+
+  --wp-state-error-100: theme('colors.int-wp-state-error.100');
+  --wp-state-neutral-100: theme('colors.int-wp-state-neutral.100');
+  --wp-state-ok-100: theme('colors.int-wp-state-ok.100');
+  --wp-state-info-100: theme('colors.int-wp-state-info.100');
+  --wp-state-warn-100: theme('colors.int-wp-state-warn.100');
+
+  --wp-hint-warn-100: theme('colors.int-wp-hint-warn.100');
+  --wp-hint-warn-200: theme('colors.int-wp-hint-warn.200');
+
+  --wp-code-100: theme('colors.int-wp-secondary.300');
+  --wp-code-200: theme('colors.int-wp-secondary.600');
+  --wp-code-text-100: theme('colors.gray.200');
+
+  --wp-link-100: theme('colors.int-wp-link.100');
+  --wp-link-200: theme('colors.int-wp-link.200');
+}
+
+:root[data-theme='dark'] {
+  --wp-background-100: theme('colors.int-wp-secondary.300');
+  --wp-background-200: theme('colors.int-wp-secondary.400');
+  --wp-background-300: theme('colors.int-wp-secondary.500');
+  --wp-background-400: theme('colors.int-wp-secondary.600');
+
+  --wp-text-100: theme('colors.gray.400');
+  --wp-text-alt-100: theme('colors.gray.500');
+
+  --wp-primary-100: theme('colors.int-wp-secondary.300');
+  --wp-primary-200: theme('colors.int-wp-secondary.400');
+  --wp-primary-300: theme('colors.int-wp-secondary.600');
+  --wp-primary-text-100: theme('colors.gray.400');
+
+  --wp-control-neutral-100: theme('colors.int-wp-secondary.300');
+  --wp-control-neutral-200: theme('colors.int-wp-secondary.600');
+  --wp-control-neutral-300: theme('colors.int-wp-secondary.700');
+
+  --wp-control-info-100: theme('colors.int-wp-control-info-dark.100');
+  --wp-control-info-200: theme('colors.int-wp-control-info-dark.200');
+  --wp-control-info-300: theme('colors.int-wp-control-info-dark.300');
+
+  --wp-control-ok-100: theme('colors.int-wp-control-ok-dark.100');
+  --wp-control-ok-200: theme('colors.int-wp-control-ok-dark.200');
+  --wp-control-ok-300: theme('colors.int-wp-control-ok-dark.300');
+
+  --wp-control-error-100: theme('colors.int-wp-control-error-dark.100');
+  --wp-control-error-200: theme('colors.int-wp-control-error-dark.200');
+  --wp-control-error-300: theme('colors.int-wp-control-error-dark.300');
+
+  --wp-state-error-100: theme('colors.int-wp-state-error-dark.100');
+  --wp-state-neutral-100: theme('colors.int-wp-state-neutral.100');
+  --wp-state-ok-100: theme('colors.int-wp-state-ok-dark.100');
+  --wp-state-info-100: theme('colors.int-wp-state-info-dark.100');
+  --wp-state-warn-100: theme('colors.int-wp-state-warn-dark.100');
+
+  --wp-hint-warn-100: theme('colors.int-wp-hint-warn-dark.100');
+  --wp-hint-warn-200: theme('colors.int-wp-hint-warn-dark.200');
+
+  --wp-code-100: theme('colors.int-wp-secondary.700');
```

Comment:
I like the color changes in general however we should change the naming / usage. I would highly prefer to use wp-secondary-700 directly instead of adding sth like `wp-code-100` as this works against the idea of tailwind / unocss / windicss to create content-full names instead of context-full.

Reply:
Yes, I agree that it is against the basic idea of the framework, but this makes it pretty hard for end users to create themes. We will end up with a list like this: https://github.com/woodpecker-ci/woodpecker/blob/6c5849c9420e105e7c1f0dc08c43040d37f3a55c/web/src/style.css

The end user does not know what `--wp-green-200` is used for, and I really think if we want to support custom color themes created by end users (https://github.com/woodpecker-ci/woodpecker/issues/1079) we should stick to semantic class names. If we decide to not implement this FR, I can just keep the color changes/harmonizing stuff and drop the CSS custom properties again. But it looks like there is still some discussion needed :smile: will put the work on hold for now.

Reply:
What can also lead to problems with user themes is the multiple use of generic classes. Just because someone wants to change the 
background color (e.g. gray.100), in most cases they will not want to change every not want to change every occurrence of gray.100.

Reply:
Still scared / pretty sure that the new custom classes will be misused in the future and therefore mess up the code, so I would personally stick to tailwind classes. But I wont block here.

Reply:
I am referring to semantic classes like `text-wp-state-error` instead of `text-red-700`. Adding extra colors / adjusting existing one like `wp-cyan-800: #...` or `red-700` with a new hex value is not a problem IMO.